src/libcrun/linux.c

```diff
@@ -4697,9 +4697,11 @@ init_container (libcrun_container_t *container, int sync_socket_container, struc
   if (UNLIKELY (ret < 0))
     return ret;
 
-  ret = libcrun_container_setgroups (container, container->container_def->process, err);
-  if (UNLIKELY (ret < 0))
-    return ret;
+  if (init_status->all_namespaces & CLONE_NEWUSER) {
+    ret = libcrun_container_setgroups (container, container->container_def->process, err);
+    if (UNLIKELY (ret < 0))
+      return ret;
+  }
```

Comment:
this must be done also when user namespaces are not used/allowed.  Can we drop this change?  What error do you get otherwise?

Reply:
``error opening file `/proc/self/setgroups`: No such file or directory``

But now that, I think about it, a fix similar to `check_running_in_user_namespace`, but in `can_setgroups` should handle it.

Reply:
untested, but I think you need something like:

```diff
diff --git a/src/libcrun/linux.c b/src/libcrun/linux.c
index e2db6ebd..cffbcb57 100644
--- a/src/libcrun/linux.c
+++ b/src/libcrun/linux.c
@@ -2942,7 +2942,16 @@ can_setgroups (libcrun_container_t *container, libcrun_error_t *err)
 
   ret = read_all_file ("/proc/self/setgroups", &content, NULL, err);
   if (ret < 0)
-    return ret;
+    {
+      /* If the file does not exist, then the kernel does not support /proc/self/setgroups and setgroups can always be used.  */
+      if (crun_error_get_errno (errno) == ENOENT)
+        {
+          crun_error_release (err);
+          return 1;
+        }
+    }
+      return ret;
+    }
 
   return strncmp (content, "deny", 4) == 0 ? 0 : 1;
 }
```

Reply:
I should have read how errors are handled in crun more. For some reason, I assumed they cannot be cleared :sweat_smile: